nixos/modules/services/networking/wireguard.nix

```diff
@@ -320,7 +320,18 @@ let
                 # cannot be used with systemd timers (see `man systemd.timer`),
                 # which is why `simple` with a loop is the best choice here.
                 # It also makes starting and stopping easiest.
+                #
+                # Retry even after "Name or service not known" dns failures:
+                Restart = "on-failure";
+                RestartSec = "${toString peer.dynamicEndpointRefreshSeconds}";
```

Comment:
I'm somewhat hesitant to introduce a whole new option for this case (partly because I don't have a straightforward way to trigger the dns failure and test the change right now). 
Why is it that your use case would benefit from such a short retry interval for dns refresh? Does the unstable network crash wireguard entirely and not "only" stop the dns refresh service?

Reply:
>  don't have a straightforward way to trigger the dns failure

I think disconnect your computer from the Internet will cause the dns failure.

> Why is it that your use case would benefit from such a short retry interval for dns refresh?

For example, if my laptop switches the wifi and `wireguard-{interface}-peer-{publickey}-refresh.service` queries dns and fails during the time when there is no network connection, I have to wait for it to rerun to re-establish a wireguard tunnel. To re-establish the wireguard tunnel as soon as possible, I want to set `RestartSec` to a small value.

Reply:
I don't think we need an extra option for that. system will only restart it 5 times in a row anyway and if someone really wants to change this they can always set systemdServices.

Reply:
What is the rationale for tying `RestartSec` to `dynamicEndpointRefreshSeconds`? The default value is `0`, which probably isn't useful here. (And it should be possible to handle start failures without enabling dynamicEndpointRefresh.)

Is the systemd default of 100ms too fast? A new option with a default of something like 5 seconds seems warranted.

Reply:
@Majiir the rationale was that I wanted to avoid the need for an additional option, but did not want to pick an arbitrary default that might be way off for some users. dynamicEndpointRefreshSeconds should approximately indicate the user's tolerance for wireguard downtime, after all. (although in the case of dns changes, not of network interruptions).

The default case of dynamicEndpointRefreshSeconds = 0 should not be a problem, because in that case the restart logic is not created.

I'd be happy with an optional dedicated option, with dynamicEndpointRefreshSeconds as the default value. Would that work for you @Majiir?

@SuperSandro2000 would such an option be acceptable?

Reply:
I see, I hadn't looked further up to see that this was all conditioned on dynamic refresh. That makes sense.

I think it would be nice to have some option for restart-on-failure without enabling the constant restarts with the dynamic refresh setting. Food for thought for another PR.